python/comps.py

```diff
@@ -193,103 +244,102 @@ def get_comps(
     return indexes_df, scores_df
 
 
-@nb.njit(fastmath=True, parallel=True)
-def _bin_by_price(observation_matrix, price_bin_matrix):
+@ti.kernel
+def _bin_by_price(
+    observation_matrix: ti.types.ndarray(),
+    price_bin_matrix: ti.types.ndarray(),
+    output_vector: ti.types.ndarray(),
+    num_observations: int,
+    num_price_bins: int
+):
     """Given a matrix of observations and a matrix of price bins, place the
     observations in the closest price bin and return an array of bin IDs
     with the same length as the observation matrix."""
-    num_observations = len(observation_matrix)
     price_bin_idx, price_bin_min_idx, price_bin_max_idx = 0, 3, 4
     observation_price_idx = 1
-    output_matrix = np.zeros(num_observations, dtype=np.int32)
 
-    for obs_idx in nb.prange(num_observations):
-        observation = observation_matrix[obs_idx]
-        observation_price = observation[observation_price_idx]
-        for bin in price_bin_matrix:
+    for obs_idx in range(num_observations):
+        observation_price = observation_matrix[obs_idx, observation_price_idx]
+        bin_found = False
```

Comment:
`for`/`else` constructs are not supported in taichi, so we have to use a flag variable instead.